src/libOpenImageIO/imagebufalgo_xform.cpp

```diff
@@ -307,13 +408,92 @@ warp_impl(ImageBuf& dst, const ImageBuf& src, const Imath::M33f& M,
 
 
 
+// Is the PV `option` in the `recognized` list? If so, return true.
+// Is it in the `obsolete` list? If so, return false.
+// Is it in neither? Return false.
+static bool
+IBA_find_optional(const ParamValue& option, cspan<ustring> recognized,
+                  cspan<ustring> obsolete = {})
+{
+    for (auto&& r : recognized)
+        if (option.name() == r)
+            return true;
+    for (auto&& o : obsolete)
+        if (option.name() == o)
+            return false;
+    return false;
+}
+
+
+
+// Given list of `options`, check if any are unrecognized or obsolete.
+// Return true if all are fine, false if not.
+static bool
+IBA_check_optional(ImageBufAlgo::KWArgs options, cspan<ustring> recognized,
+                   cspan<ustring> obsolete = {})
+{
+    bool ok = true;
+    for (auto&& pv : options) {
+        ok &= IBA_find_optional(pv, recognized, obsolete);
+    }
+    return ok;
+}
+
+
+
+bool
+ImageBufAlgo::warp(ImageBuf& dst, const ImageBuf& src, M33fParam M,
+                   KWArgs options, ROI roi, int nthreads)
+{
+    static ustring recognized[] = { "filtername"_us,    "filterwidth"_us,
+                                    "wrap"_us,          "edgeclamp"_us,
+                                    "recompute_roi"_us, "filterptr"_us };
+    IBA_check_optional(options, recognized);
+
+    Filter2D::ref filterref;
+    if (options.contains("filterptr", TypePointer)) {
+        auto f = options.find("filterptr", TypePointer);
```

Comment:
Why `contains` (which under the hood I think calls `find`) followed by `find`? Would it be simpler and faster to just do the `find` and then the `OIIO_DASSERT` code is what is used for the actual condition logic?

Reply:
I've rearranged this better.